behavior:
  head:
    # Name of the head_behaviors ros-node
    rosnode: head_behavior

    # Sane default values for some modules
    defaults:
      head_mode: 0    # Ball mode

    # TODO maybe keep this instead of the generated lines
    # Search pattern for penalty
    #search_pattern_penalty:
    #  - [0, -15]
    #  - [0, -30]
    #  - [30, -7]
    #  - [-30, -7]

    # Search pattern for ball
    search_pattern:
        # Speed search pattern
        tilt_speed: 2
        pan_speed: 2

        # Max values for the search pattern
        pan_max: [120, -120]
        tilt_max: [0, -70]

        # Number of scan lines for the search pattern
        scan_lines: 4

    # Search pattern for penalty
    search_pattern_penalty:
        tilt_speed: 1
        pan_speed: 2
        pan_max: [-30, 30]
        tilt_max: [-7, -30]
        scan_lines: 2

    # Search pattern for visual compass features search
    visual_compass_features_pattern:
        tilt_speed: 1
        pan_speed: 1
        pan_max: [-45, 45]
        tilt_max: [7, -7]
        scan_lines: 2

    search_pattern_goal:
        tilt_speed: 1
        pan_speed: 1.5
        pan_max: [120, -120]
        tilt_max: [7, -7]
        scan_lines: 2

    search_pattern_field_features:
        tilt_speed: 1
        pan_speed: 1
        pan_max: [-90, 90]
        tilt_max: [7, 0]
        scan_lines: 2

    search_recent_ball:
        tilt_speed: 2
        pan_speed: 2
        offset_pattern: [
            [ 0, -10],
            [ 10,-10],
            [ 10, 10],
            [-10, 10],
            [-10,-20],
            [ 20,-20]
        ]
        ball_search_time: 10

    # Values for the visual compass record pattern
    record_pattern_scan_lines: 2
    record_pattern_pan_max_left: [-80, 0]
    record_pattern_pan_max_right: [0, -80]
    record_pattern_tilt_max: [14, 0]

    # Visual compass ground truth trigger topic
    visual_compass_trigger_topic: '/visual_compass_ground_truth_trigger'

    # These values describe the minimal required delta between current joint states and target joint states in degrees
    # to reduce unnecessary movement due to noise in the detection of the BallRelative.
    ball_tracking_min_pan_delta: 3
    ball_tracking_min_tilt_delta: 3

    # Positions for static head modes
    look_down_position: [0, -65]
    look_forward_position: [0, -7]
    look_up_position: [0, 12]

    # Threshold (in degrees) when a head position is reached and
    # the next position will be triggered
    position_reached_threshold: 5

    # After `ball_lost_time` seconds, the ball is considered lost and will be searched
    ball_lost_time: 2
    post_lost_time: 1
